internal/protocol/download_manager.go

```diff
@@ -355,6 +355,54 @@ func (m *BlockDownloadManager) rescan(ctx context.Context) {
 	// TODO:  Expire obsolete entries (behind LIB)
 }
 
+func (m *BlockDownloadManager) clearPeerRequests(peerID peer.ID) int {
+	// When a peer closes, we have to delete all of its in-flight blocks
+	// from Downloading (this solves issue #151).
+
+	numDeleted := 0
+
+	toClear := make([]util.BlockTopologyCmp, 0)
+	for k, v := range m.Downloading {
+		if v.PeerID == peerID {
+			toClear = append(toClear, k)
+		}
+	}
+	numDeleted += len(toClear)
+	for _, k := range toClear {
+		delete(m.Downloading, k)
+	}
+
+	// I don't think it's necessary to clear Applying / WaitingToApply,
+	// as fully downloaded blocks can continue to be processed
+	// even when the peer is disconnected.  For now I'll leave the
+	// commented code here in case we decide to do this at some
+	// point in the future.
+	/*
```

Comment:
Is this code no longer necessary? If so, it should just be removed,

Reply:
Issue #155 is the issue for dealing with this code.  I updated the comment to suggest that the code should be either uncommented or removed as part of the resolution of that issue.